source/Ship.cpp

```diff
@@ -1335,6 +1346,20 @@ void Ship::SetGovernment(const Government *government)
 
 
 
+void Ship::SetIsPlayerFlagship(bool isFlagship)
+{
+	this->isPlayerFlagship = isFlagship;
+}
+
+
+
+void Ship::SetFirePrimary(bool firePrimary)
+{
+	this->firePrimary = firePrimary;
```

Comment:
Shouldnt this be renamed to firingprimaries since its a state and not an action?

Reply:
The value is meant to indicate that the Command::PRIMARY is set (by the player) i.e. the player wants to fire their primary weapons. There might be some animation that needs to happen before the primary weapons actually fire, so it is not necessarily true that the primary weapons will be firing when this value is true.

I do want to rethink this implementation though, it does seem a bit haphazard
See comment: https://github.com/endless-sky/endless-sky/pull/7150#discussion_r1041499812